### PR TITLE
fix(docker,jobs): scope container cleanup to owner labels and preserve pause/cancel status

### PR DESCRIPTION
**Pull request by ironcommit**
## Docker: scope container cleanup and orphan reconciliation to owner labels

Add `models_docker_container_labels` config field to `DockerBackendConfig` so containers created by the models controller carry arbitrary labels. Integration test fixtures use this to stamp containers with test-run and test-worker identity labels, scoping teardown cleanup and orphan reconciliation to the owning pytest-xdist worker.

- New `_managed_container_labels()` helper on the creation reconciler centralises label construction for NIM, sidecar, and puller containers
- `cleanup_model_deployment_containers()` accepts optional owner labels and only removes matching containers
- Pass `ignore_removed=True` on Docker container list calls to prevent races when another worker removes a container during enumeration
- Diagnostic output now includes owner labels for easier debugging

## Jobs: preserve pause/cancel status when pods report task errors

When a Kubernetes job is pausing, paused, cancelling, or cancelled, pod-level task errors (e.g. SIGTERM exit code 137) no longer override the job status to ERROR. This prevents user-initiated pause/cancel from incorrectly surfacing as a failure.

## Tests: fix intake ATIF ingest timestamp

Replace the frozen `_BASE_TIME` (2026-01-15) with a dynamic timestamp 7 days in the past so ingested spans stay within ClickHouse's 90-day TTL and tests don't start failing as time passes.

## Summary by CodeRabbit

* **New Features**
  * Added configurable `model_labels` support for applying custom labels to model resources across Docker and Kubernetes backends.
  * Test runs now tag model resources with run-/worker-specific labels to enable scoped teardown and better traceability.
* **Bug Fixes**
  * Docker cleanup now stops/removes only containers that match the expected label set; orphan detection and port selection ignore containers already marked removed.
  * Kubernetes job reconciliation now preserves `PAUSED`/`CANCELLED` when task errors occur during terminal transitions.
* **Documentation**
  * Updated the configuration reference with the new Docker Models service label option.
* **Tests**
  * Added unit and regression test coverage for label scoping and reconciliation edge cases.

### CODE REVIEW

**Review by maxdubrinsky — APPROVED** [review]

File: docs/set-up/config-reference.mdx [hunk 1]
```diff
@@ -577,6 +579,8 @@ models:
         default_tolerations:
         # Default Kubernetes node selector for all NIM deployments. Can be overridden per-deployment via k8s_nim_operator_config. Example: {'node-type': 'gpu-node', 'zone': 'us-west1-a'}
         default_node_selector:
+        # Additional labels copied onto model resources managed by the models controller.
+        model_labels:
```
Comment (mckornfield) [inline docs/set-up/config-reference.mdx:583]: hmm why does this not have `{}` after? I guess it's null but looks weird

File: services/core/jobs/src/nmp/core/jobs/controllers/backends/kubernetes/kubernetes_job.py [hunk 2]
```diff
@@ -311,7 +311,26 @@ def create_step_update(self, step: PlatformJobStepWithContext, job: V1Job) -> Jo
             error_details["message"] = status_details.get("message", "Job encountered an error")
         status_details["events"] = self.get_kube_job_events(job)
         task_has_error = update_all_tasks(self._nmp_sdk, self._core_v1, self.namespace, step)
-        if task_has_error:
+        teardown_lifecycle_statuses = {
```
Comment (mckornfield) [inline services/core/jobs/src/nmp/core/jobs/controllers/backends/kubernetes/kubernetes_job.py:314]: should this be a constant/could it be used anywhere else?

**Review by mckornfield — APPROVED** [review]

File: services/core/jobs/src/nmp/core/jobs/controllers/backends/kubernetes/kubernetes_job.py (hunk 2 above)
Comment (ironcommit) [reply services/core/jobs/src/nmp/core/jobs/controllers/backends/kubernetes/kubernetes_job.py:314]: It's not used anywhere else at the moment.

File: docs/set-up/config-reference.mdx (hunk 1 above)
Comment (ironcommit) [reply docs/set-up/config-reference.mdx:583]: yeah, does look wierd.  The first one does have it, but the second one follows the convention around it which is null.
